spring-integration-core/src/test/java/org/springframework/integration/dsl/transformers/TransformerTests.java

```diff
@@ -287,6 +342,71 @@ public IdempotentReceiverInterceptor idempotentReceiverInterceptor() {
 			return idempotentReceiverInterceptor;
 		}
 
+		@Bean
+		public PollableChannel subFlowTestReplyChannel() {
+			return new QueueChannel();
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow syncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle((p, h) -> someServiceGateway.deriveValueSync(p)))
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow asyncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
```

Comment:
Not sure why do you worry about `async` behavior in the sub-flow, since `ContentEnricher` uses blocking `sendAndReceiveMessage()`. So, it doesn't matter how the reply is going to appear. But yeah, it won't hurt to have that test as well. 
M-m-m. I believe that `@Autowired` is redundant on the `@Bean` method. Any its arguments are really considered for autowiring.